core/contract/bridge/cross_uri.go

```diff
@@ -0,0 +1,110 @@
+package bridge
+
+import (
+	"fmt"
+	"net/url"
+
+	pb "github.com/xuperchain/xuperchain/core/contractsdk/go/pb"
+	xchainpb "github.com/xuperchain/xuperchain/core/pb"
+)
+
+// TODO:zq
+
+const (
+	// XuperScheme define the xuper scheme
+	XuperScheme = "xuper"
+)
+
+// CrossChainURI Standard
+// [scheme:][//chain_name][path][?query]
+// eg xuper://chain1?module=wasm&bcname=xuper&contract_name=counter&method_name=increase
+type CrossChainURI struct {
+	*url.URL
+}
+
+// ParseCrossChainURI will parse uri to cross chain uri instance
+func ParseCrossChainURI(crossChainURI string) (*CrossChainURI, error) {
+	uri, err := url.Parse(crossChainURI)
+	if err != nil {
+		return nil, err
+	}
+
+	return &CrossChainURI{
+		uri,
+	}, nil
+}
+
+// GetScheme return cross chain uri scheme
+func (ccu *CrossChainURI) GetScheme() string {
+	return ccu.URL.Scheme
+}
+
+// GetChainName return cross chain uri chain name
+func (ccu *CrossChainURI) GetChainName() string {
+	return ccu.URL.Host
+}
+
+// GetPath return cross chain uri path
+func (ccu *CrossChainURI) GetPath() string {
+	return ccu.URL.Path
+}
+
+// GetQuery return cross chain uri query
+func (ccu *CrossChainURI) GetQuery() url.Values {
+	return ccu.URL.Query()
+}
+
+// CrossChainScheme define the interface of CrossChainScheme
+type CrossChainScheme interface {
+	GetCrossQueryRequest(*CrossChainURI, []*pb.ArgPair, string, []string) (*xchainpb.CrossQueryRequest, error)
```

Comment:
It is better to add names to interface parameters.

Reply:
Thanks, fixed